resource/ui/ts/models/log.ts

```diff
@@ -0,0 +1,149 @@
+        var tdStyleEvenFirst = "font-family:Arial, sans-serif;font-size:14px;padding:10px 5px;border-style:solid;border-width:1px;overflow:hidden;word-break:normal;border-color:#ccc;color:#333;background-color:#efefef;text-align:center";
+        var tdStyleEven = "font-family:Arial, sans-serif;font-size:14px;padding:10px 5px;border-style:solid;border-width:1px;overflow:hidden;word-break:normal;border-color:#ccc;color:#333;background-color:#fff;text-align:center";
+
+        export class Entries {
+
+            startTime = Utils.chainProp(this, <number>null);
+            endTime = Utils.chainProp(this, <number>null);
+            max = Utils.chainProp(this, <number>null);
+            level = Utils.chainProp(this, <string>null);
+
+            private _url():string {
+                var url = "/_status/local/log";
+                if (this.level() != null) {
+                    url += "/" + this.level();
+                }
+                url += "?";
+                if (this.startTime() != null) {
+                    url += "startTime=" + this.startTime().toString() + "&";
+                }
+                if (this.endTime() != null) {
+                    url += "entTime=" + this.endTime().toString() + "&";
+                }
+                if (this.max() != null) {
+                    url += "max=" + this.max().toString() + "&";
+                }
+                return url;
+            }
+
+            private _data = new Utils.QueryCache((): promise<Proto.LogEntry[]> => {
+                return m.request({ url: this._url(), method: "GET", extract: nonJsonErrors })
+                    .then((results: LogResponseSet) => {
+                        return results.d;
+                    });
+            })
+
+            public Refresh() {
+                this._data.refresh();
+            }
+
+            public Entries(): Proto.LogEntry[] {
+                return this._data.result();
+            }
+
+            private static _formatTags = new RegExp("%s|%d|%v|%+v", "gi")
+            private static _FormatMessage(entry: Proto.LogEntry): string {
+                var i = -1;
+                return entry.format.replace(Entries._formatTags, function(tag) {
+                    i++;
+                    if (entry.args.length > i) {
+                        return entry.args[i].str;
+                    } else {
+                        return "";
+                    }
+                });
+            }
+
+            private static _EntryRow(entry: Proto.LogEntry, count: number): _mithril.MithrilVirtualElement {
+                var dstyle, countStyle: string;
+                if (count % 2 == 0) {
+                    countStyle = tdStyleEvenFirst;
+                    dstyle = tdStyleEven;
+                } else {
+                    countStyle = tdStyleOddFirst;
+                    dstyle = tdStyleOdd;
+                }
+                return m("tr", [
```

Comment:
This is View code, not Model code.  This was done on the Nodes and Status pages as well - the models contain rendering code.

What should really be done is to create a component that renders tabular data.  In the case of Node and Status pages tables were not appropriate, but in the case of logs there are multiple data columns, so I can buy a justification for using tables.